library/src/main/kotlin/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadAdapter.kt

```diff
@@ -226,10 +240,182 @@ internal sealed class TransactionPayloadViewHolder(view: View) : RecyclerView.Vi
+
+        private fun JsonArray.showArrayObjects() {
+            map {
+                TransactionPayloadItem.BodyCollapsableItem(jsonElement = it.asJsonObject)
+            }.also { list ->
+                with(bodyBinding) {
+                    imgExpand.gone()
+                    txtKey.gone()
+                    txtDivider.gone()
+                    txtStartValue.gone()
+                    txtEndValue.gone()
+                    rvSectionData.show()
+                    rvSectionData.adapter = TransactionBodyAdapter().also { adapter ->
+                        adapter.setItems(list)
+                    }
+                }
+            }
+        }
+
+        private fun View.setClickForValue(element: JsonElement) = with(bodyBinding) {
+            var isOpen = false
+
+            imgExpand.show()
+            txtStartValue.text = if (element.isJsonObject) "{...}" else "[...]"
+            txtEndValue.gone()
+
+            setOnClickListener { view ->
+                isOpen = isOpen.not()
+
+                imgExpand.animate()
+                    .rotationBy(if (isOpen) OPEN_ROTATION_VALUE else CLOSE_ROTATION_VALUE)
+                    .setListener(object : Animator.AnimatorListener {
+                        override fun onAnimationStart(p0: Animator) {
+                            view.isClickable = false
+                        }
+
+                        override fun onAnimationEnd(p0: Animator) {
+                            view.isClickable = true
+
+                            if (isOpen) {
+                                rvSectionData.show()
+                                txtStartValue.text = if (element.isJsonObject) "{" else "["
+                                txtEndValue.show()
+                                txtEndValue.text = if (element.isJsonObject) "}," else "],"
+                            } else {
+                                rvSectionData.gone()
+                                txtStartValue.text =
+                                    if (element.isJsonObject) "{...}," else "[...],"
+                                txtEndValue.gone()
+                            }
+
+                            rvSectionData.adapter = TransactionBodyAdapter().also { adapter ->
+                                adapter.setItems(
+                                    listOf(TransactionPayloadItem.BodyCollapsableItem(jsonElement = element))
+                                )
+                            }
+                        }
+
+                        @Suppress("EmptyFunctionBlock")
+                        override fun onAnimationCancel(p0: Animator) {
+                        }
+
+                        @Suppress("EmptyFunctionBlock")
+                        override fun onAnimationRepeat(p0: Animator) {
+                        }
+                    })
+            }
+        }
+
+        private fun View.show() = apply { isVisible = true }
+        private fun View.gone() = apply { isVisible = false }
```

Comment:
There is a ktx library inside androidx that was providing those functions already.

```
view.isGone = true
view.isVisible = true
```

Can we use it?